docs/source/user/theory.rst

```diff
@@ -0,0 +1,890 @@
+   %  \draw (0, \i*\dy, 0) -- (\nx*\dx, \i*\dy, 0);
+   %}
+   \end{scope}
+
+   % second diagram (shifted to the right)
+   \begin{scope}[xshift=5cm]  % Adjust spacing between the two diagrams
+
+   % parameters
+   \def\dx{0.1}
+   \def\dy{0.1}
+   \def\dz{0.1}
+   \def\nx{31}
+   \def\ny{7}
+   \def\nz{3}
+
+   % colours
+   \definecolor{redw}{rgb}{1,0.45,0.45}
+   \definecolor{bluew}{rgb}{0.45,0.45,1}
+
+   % sub-groups
+   \foreach \i in {1,...,8} {
+     \foreach \x in {0,...,31} {
+       % group labels
+       \node[](Group\i) at (-0.3, 1.5*8*4*\dy - 0.2 - \i*4*\dy - \i*0.1) {\i};
+
+       % grid points
+       \foreach \y in {0,...,3} {
+         \foreach \z in {0} {
+           \pgfmathsetmacro\k{\x*2.5}
+           \draw [color=red!\k!blue, mark=*, mark size=1]
+                 plot coordinates {(\x*\dx, \y*\dy + \i*4*\dy + \i*0.1, \z*\dz)};
+         }
+       }
+     }
+   }
+   \end{scope}
+
+Typically, for a double-precision simulation on CPUs, :math:`SZ=8` as vector registers (512 bits) handle 8 double-precision FLOPs per cycle. For GPUs, :math:`SZ=32` as a single streaming multiprocessor (SM) typically has 32 double-precision cores in total, where each core is effectively assigned an individual tridiagonal system.
+
+Tridiagonal systems resulting from high-order compact finite-difference schemes are always diagonally dominant, and the algorithm used in x3d2 takes advantage of this characteristic to minimise communication requirements between ranks. This involves dividing a batch of tridiagonal systems into multiple domains, where the subdomains are located across multiple ranks in a distributed memory environment. 
+
+Apart from the specialised data structure in DistD2, another novel aspect of this algorithm is that it fuses the RHS construction based on discretised equations such as Eq. :eq:`first-derivative` with the forward pass in the decoupling phase of the algorithm -- this minimises data movement requirements.
+
+After the decoupling phase of the algorithm, the next step is to solve the reduced system. By exploiting diagonal dominance, certain entries in the reduced system  can be eliminated without any loss in numerical accuracy, leading to independent :math:`2 \times 2` systems that are coupled across MPI boundaries. This step is the main deviation from other tridiagonal solvers that use a PCR-type strategy to obtain the solution of the reduced system on a single rank. In the DistD2 algorithm, the reduced systems are not coupled across the entire domain, but only across 2 individual ranks across an MPI boundary. These smaller systems are then solved efficiently using local communication between neighbouring ranks. This single-step MPI communication involving two neighbouring ranks is a significant advantage of the DistD2 algorithm over existing strategies. Finally, the substitution phase of the algorithm requires a simple algebraic substitution and does not require any further MPI communications.
+
```

Comment:
Just to mark that I have reviewed 'Tridiagonal systems' section up to this point.